conformance_tests/sysman/test_sysman_init/src/test_init_sysman.cpp

```diff
@@ -24,4 +24,51 @@ TEST(SysmanInitTests,
   ASSERT_GT(pCount, 0);
 }
 
+TEST(SysmanInitTests,
+     GivenCoreAndSysmanInitializedSysmanHandleWorksForEnumFrequencyDomanis) {
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  uint32_t driver_count = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesDriverGet(&driver_count, nullptr));
+  ASSERT_GT(driver_count, 0);
+  std::vector<zes_driver_handle_t> drivers(driver_count);
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesDriverGet(&driver_count, drivers.data()));
+
+  uint32_t device_count = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesDeviceGet(drivers[0], &device_count, nullptr));
+  std::vector<zes_device_handle_t> devices(device_count);
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesDeviceGet(drivers[0], &device_count, devices.data()));
+
+  uint32_t count = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesDeviceEnumFrequencyDomains(devices[0], &count, nullptr));
+
+}
+
+TEST(SysmanInitTests,
+     GivenCoreAndSysmanInitializedCoreHandleReturnsUninitializedErrorForEnumFrequencyDomanis) {
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  uint32_t driver_count = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeDriverGet(&driver_count, nullptr));
+  ASSERT_GT(driver_count, 0);
+  std::vector<zes_driver_handle_t> drivers(driver_count);
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeDriverGet(&driver_count, drivers.data()));
+
+  uint32_t device_count = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeDeviceGet(drivers[0], &device_count, nullptr));
+  std::vector<zes_device_handle_t> devices(device_count);
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeDeviceGet(drivers[0], &device_count, devices.data()));
+
+  uint32_t count = 0;
+  ASSERT_EQ(ZE_RESULT_ERROR_UNINITIALIZED, zesDeviceEnumFrequencyDomains(devices[0], &count, nullptr));
```

Comment:
Is this coming as ZE_RESULT_ERROR_UNINITIALIZED here?

Reply:
If we do not do zesInit then it returns ZE_RESULT_ERROR_UNINITIALIZED. If zesInit is done then there is a seg fault. Issue [NEO-12690](https://jira.devtools.intel.com/browse/NEO-12690) gets reproduced in this. Will wait for the fix.